### PR TITLE
API2000 support for LI

### PR DESCRIPTION
**Pull request by yuvirani**
### Description
API2000 support for LI

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
  - [x] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [x] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: examples/logical_interconnect.rb [hunk 1]
```diff
@@ -41,24 +44,42 @@
   enclosure 'Encl1'
   bay_number 1
   action :add_interconnect
+  only_if { client[:api_version] == 200 }
+end
+
+# Get the inconsistency report for bulk update from group
```
Comment (VenkateshRavula) [inline examples/logical_interconnect.rb:50]: add if condition for this example (if api >= 2000)

File: endpoints-support.md [hunk 2]
```diff
@@ -100,36 +100,36 @@
...
+|<sub>/rest/logical-interconnects/{id}/telemetry-configurations/{tcId}</sub>              | PUT       | :white_check_mark:   | :white_check_mark:   |  :white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   | :white_check_mark:   | :white_check_mark:   |:white_check_mark:   |
+|<sub>/rest/logical-interconnects/compliance</sub>                                        | PUT      | :heavy_multiplication_x:       | :heavy_multiplication_x: | :heavy_multiplication_x: | :heavy_multiplication_x: | :heavy_multiplication_x: |:heavy_multiplication_x: |:heavy_multiplication_x:   | :heavy_multiplication_x: |:heavy_multiplication_x: |:heavy_multiplication_x: |
+|<sub>/rest/logical-interconnects/{id}</sub>                                              | PATCH     | :white_check_mark:   | :white_check_mark:   | :heavy_minus_sign:   |:heavy_minus_sign:   |:heavy_minus_sign:   |:heavy_minus_sign:   |:heavy_minus_sign:  | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   |
```
Comment (VenkateshRavula) [inline endpoints-support.md:132]: Also add endpoint for "bulk_inconsistency_validate"

File: spec/fixtures/cookbooks/oneview_test_api2000_c7000/recipes/logical_interconnect_bulk_inconsistency_validate.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Cookbook Name:: oneview_test
```
Comment (VenkateshRavula) [inline spec/fixtures/cookbooks/oneview_test_api2000_c7000/recipes/logical_interconnect_bulk_inconsistency_validate.rb]: Cookbook name should be "oneview_test_api2000_c7000"

File: spec/fixtures/cookbooks/oneview_test_api2000_c7000/recipes/logical_interconnect_bulk_inconsistency_validate.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+# Recipe:: logical_interconnect_update_bulk_inconsistency_validate
```
Comment (VenkateshRavula) [inline spec/fixtures/cookbooks/oneview_test_api2000_c7000/recipes/logical_interconnect_bulk_inconsistency_validate.rb]: Change Recipe name to "logical_interconnect_bulk_inconsistency_validate"

**Comment by VenkateshRavula** [comment]
Update Berks file with new cookbook name.
Refer https://github.com/HewlettPackard/oneview-chef/pull/449/files

File: spec/unit/resources/logical_interconnect/bulk_inconsistency_validate_spec.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+require_relative './../../../spec_helper'
+
+describe 'oneview_test_api2000_synergy::logical_interconnect_bulk_inconsistency_validate' do
```
Comment (VenkateshRavula) [inline spec/unit/resources/logical_interconnect/bulk_inconsistency_validate_spec.rb]: you have mentioned cookbook name as "oneview_test_api2000_c7000". So mention the same here.

File: libraries/resource_providers/api2000/c7000/logical_interconnect_provider.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+module OneviewCookbook
+  module API2000
+    module C7000
+      # LogicalInterconnect API2000 C7000 provider
+      class LogicalInterconnectProvider < OneviewCookbook::API1800::C7000::LogicalInterconnectProvider
+        def bulk_inconsistency_validate
+          @item['uri'] = @new_resource.logicalInterconnectUris[0]
```
Comment (VenkateshRavula) [inline libraries/resource_providers/api2000/c7000/logical_interconnect_provider.rb:18]: what does this [0] mean..?

Comment (yuvirani) [reply libraries/resource_providers/api2000/c7000/logical_interconnect_provider.rb:18]: Taking resource uri from the input

File: endpoints-support.md [hunk 7]
```diff
@@ -100,36 +100,37 @@
...
+|<sub>/rest/logical-interconnects/bulk-inconsistency-validation                           | POST       | :heavy_minus_sign:  |:heavy_minus_sign:  |:heavy_minus_sign:  |:heavy_minus_sign:  |:heavy_minus_sign:  |:heavy_minus_sign:  |:heavy_minus_sign:  |:heavy_minus_sign:  |:heavy_minus_sign:  |:heavy_minus_sign:  |
```
Comment (VenkateshRavula) [inline endpoints-support.md:133]: make it white_check_mark for api2000

**Pull request closed without merging**